flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
         const_dir : float, optional
             direction of time-invariant wind forcing [deg], by default None
         """
-        self.sf_model.setup_wind_forcing(
-            timeseries=timeseries,
-            magnitude=const_mag,
-            direction=const_dir,
+        if isinstance(forcing, WindConstant):
+            self._model.setup_wind_forcing(
+                timeseries=None,
+                const_mag=forcing.speed,
+                const_dir=forcing.direction,
+            )
+        elif isinstance(forcing, WindTimeSeries):
+            self._model.setup_wind_forcing(
+                timeseries=forcing.path, const_mag=None, const_dir=None
+            )
+        elif isinstance(forcing, WindFromModel):
+            self._model.setup_wind_forcing_from_grid(wind=forcing.path)
+        else:
+            self._logger.warning(
+                f"Unsupported wind forcing type: {forcing.__class__.__name__}"
+            )
+            return
+
+    def _add_forcing_rain(self, forcing: IRainfall):
+        """Add spatially constant rain forcing to sfincs model. Use timeseries or a constant magnitude.
+
+        Parameters
+        ----------
+        timeseries : Union[str, os.PathLike], optional
+            path to file of timeseries file (.csv) which has two columns: time and precipitation, by default None
+        const_intensity : float, optional
+            time-invariant precipitation intensity [mm/hr], by default None
+        """
+        if isinstance(forcing, RainfallConstant):
+            self._model.setup_precip_forcing(
+                timeseries=None,
+                magnitude=forcing.intensity,
+            )
+        elif isinstance(forcing, RainfallSynthetic):
+            self._model.setup_precip_forcing(
+                timeseries=forcing.path,
+                magnitude=None,
+            )
+        elif isinstance(forcing, RainfallFromModel):
+            # TODO: check how to add this to the model
+            data = self._get_wl_df_from_offshore_his_results()
+            self._model.setup_precip_forcing_from_grid(precip=data)
+
+        elif isinstance(forcing, RainfallFromSPWFile):
+            # TODO: check how to add this to the model
```

Comment:
These should be passed as a keyword when the spiderweb file is created. So for this we cannot separate wind and rain forcing that neatly. Sorry!

Reply:
Im not sure what you mean exactly with passing a keyword, lets discuss tomorrow/Wednesday!